src/conn/conn_api.c

```diff
@@ -1641,6 +1651,10 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
         bytelock = false;
         ret = 0;
     }
+    if (!F_ISSET(conn, WT_CONN_READONLY) && ret == ENOENT) {
```

Comment:
I think (given the `if` statement that precedes this one), that the check for `!F_ISSET(conn, WT_CONN_READONLY)` is redundant.  Secondly, this seems kind of subtle to me - how does the lock file not existing imply corruption?  Probably needs a comment?